### PR TITLE
Dynamic loading

### PR DESCRIPTION
**Pull request by Wenzel**
Add runtime dynamic loading for `libxenctrl`

cc @rageagainsthepc

### CODE REVIEW

**Comment by Wenzel** [comment]
No more xen libraries linking with libmicrovmi:
~~~
[root@localhost libmicrovmi]# ldd target/debug/examples/pause
	linux-vdso.so.1 (0x00007ffcf89e9000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007fbeb10a5000)
	librt.so.1 => /lib64/librt.so.1 (0x00007fbeb109b000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007fbeb107a000)
	libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007fbeb105f000)
	libc.so.6 => /lib64/libc.so.6 (0x00007fbeb0e99000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fbeb148d000)

~~~

**Comment by Wenzel** [comment]
helps fix https://github.com/Wenzel/libmicrovmi/issues/68

File: xenctrl.iml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rageagainsthepc) [inline xenctrl.iml]: Are you sure you want to check this in?

File: src/error.rs [hunk 2]
```diff
@@ -1,37 +1,37 @@
...
+        Self { desc: s.to_owned() }
     }
 
-    pub fn desc(self) -> &'static str {
-        unsafe {
-            let desc = xc_error_code_to_desc(self.0 as _);
-            ffi::CStr::from_ptr(desc).to_str().unwrap()
-        }
-    }
+    // TODO: call xc_error_code_to_desc using libxenctrl
```
Comment (rageagainsthepc) [inline src/error.rs]: I would just remove this then...

File: src/error.rs [hunk 3]
```diff
@@ -1,37 +1,37 @@
...
-    error, ffi,
+    error,
     fmt::{self, Debug, Display, Formatter},
 };
 
-use xenctrl_sys::{xc_error_code, xc_error_code_to_desc};
-
-#[derive(Copy, Clone)]
-pub struct Error(xc_error_code);
+pub struct Error {
```
Comment (rageagainsthepc) [inline src/error.rs]: I think it is a little bit confusing that the error struct shadows the name of the general error interface.

File: xenctrl.iml (hunk 1 above)
Comment (Wenzel) [reply xenctrl.iml]: good catch.
that's why the CI failed at first.
it said a file was missing, and I was sure I added `libxenctrl.rs` in the git.
but the file I added was this one :-)

File: src/error.rs (hunk 3 above)
Comment (Wenzel) [reply src/error.rs]: Right, we should renamed it `XcError`

**Review by rageagainsthepc — COMMENTED** [review]
Lgtm so far.

**Comment by Wenzel** [comment]
thanks for the review

**Pull request merged**